templates/spa.php

```diff
@@ -0,0 +1,235 @@
+                    </aside>
+
+                    <div class="space-y-6 sm:px-6 lg:px-0 lg:col-span-9">
+                        <div id="cookies">
+                            <div class="shadow sm:rounded-md sm:overflow-hidden">
+                                <div class="bg-white">
+                                    <div class="px-4 py-5 sm:px-6 bg-gray-100">
+                                        <h3 class="text-lg leading-6 font-medium text-gray-900">Example API Response</h3>
+                                    </div>
+
+                                    <div class="px-4 py-5 sm:px-6 grid grid-cols-6 gap-6">
+                                        <div class="col-span-6">
+                                            <textarea id="backendApiResponse" readonly="readonly" class="mt-1 block w-full bg-white border border-gray-300 rounded-md shadow-sm py-2 px-3 focus:outline-none focus:ring-indigo-500 focus:border-indigo-500 sm:text-sm"></textarea>
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </main>
+
+        <script>
+            createAuth0Client({
+                domain: '<?php echo $config->getDomain(); ?>',
+                client_id: '<?php echo $config->getClientId(); ?>',
+                audience: '<?php echo $config->buildDefaultAudience(); ?>',
+                redirect_uri: '<?php echo $router->getUri('/', ''); ?>',
+                scope: '<?php echo $config->buildScopeString() ?>'
+            }).then(auth0 => {
+                document.getElementById('login').addEventListener('click', () => {
+                    auth0.loginWithRedirect().catch(() => {});
+                });
+
+                document.getElementById('logout').addEventListener('click', () => {
+                    auth0.logout();
+                });
+
+                auth0.getUser().then(user => {
+                    if (user) {
+                        auth0.getUser().then(user => {
+                            setAppState(true, user);
+
+                            // Issue a request to our quickstart backend api at /api
+                            callBackendApi(auth0);
+                        });
+
+                        return;
+                    }
+
+                    auth0.handleRedirectCallback().then(redirectResult => {
+                        console.log(redirectResult);
+
+                        auth0.getUser().then(user => {
+                            setAppState(true, user);
+
+                            // Issue a request to our quickstart backend api at /api
+                            callBackendApi(auth0);
+                        });
+                    }).catch(() => {
+                        setAppState(false);
+                        callBackendApi();
+                    });
```

Comment:
Rather than wrapping this in a check to `auth.getUser`, you'd be better wrapping it in a check to see if you actually need to handle an auth callback. The way we do this in our SDKs is by checking for `state && (code || error)` in `window.location.search` (here's [an example](https://github.com/auth0/auth0-react/blob/67670325585a2994d381426cdf9295d92695b1ae/src/utils.tsx#L7-L9)).

Otherwise if you're not logged in, it will just try and handle the callback every time when it doesn't need to (it might even through warnings in the console).

Reply:
Good call, pushing that improvement